modules/oe_corporate_countries_address/tests/src/Kernel/AvailableCountriesSubscriberTest.php

```diff
@@ -0,0 +1,310 @@
+   */
+  protected $bundleName = 'oe_countries';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('configurable_language');
+    $this->installConfig(['language']);
+    // At least one extra language needs to be enabled for translations to work.
+    ConfigurableLanguage::createFromLangcode('it')->save();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->state = $this->container->get('state');
+
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('entity_test');
+
+    $entity = EntityTest::create();
+
+    FieldStorageConfig::create([
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'type' => $this->fieldTypeName,
+    ])->save();
+
+    FieldConfig::create([
+      'label' => 'My field',
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'bundle' => $this->entityTypeName,
+    ])->save();
+
+    $entity_form_display = EntityFormDisplay::collectRenderDisplay($entity, 'default');
+
+    $entity_form_display->setComponent($this->fieldName, [
+      'region' => 'content',
+      'type' => 'address_country_default',
+    ]);
+    $entity_form_display->save();
+
+  }
+
+  /**
+   * Tests the deprecated countries are removed from the countries array.
+   * */
+  public function testGetAvailableCountriesArray(): void {
+
+    $values = [
+      $this->fieldName => [
+        'value' => 'FR',
+      ],
+    ];
+
+    $entity = EntityTest::create($values);
+    $entity->save();
+
+    $entity = $this->reloadEntity($entity);
+
+    /** @var \Drupal\field\Entity\FieldStorageConfig $field_storage */
+    $field_storage = $entity->getFieldDefinitions()[$this->fieldName]
+      ->getFieldStorageDefinition();
+
+    $field_storage->getPropertyDefinitions();
+
+    // Get available_countries from field.
+    // This doesn't have the deprecated countries.
+    $form = $this->buildEntityTestForm();
+    $availableCountries = $form["field_country"]["widget"][0]["value"]["#available_countries"];
+
+    // Get all the countries including the deprecated countries from sparql db.
+    $allCountries = $this->getAllCountries();
+    // Get deprecated countries from sparql db.
+    $countriesDeprecated = $this->getDeprecatedCountries($allCountries);
+
+    // Test: Assert that our options does not have deprecated countries.
+    $result = $this->hasDeprecatedCountry(array_keys($availableCountries), $countriesDeprecated);
+    $this->assertEquals(FALSE, $result, 'Available countries does not have deprecated countries');
+
+    // Test: Assert that available countries has the same countries
+    // as allCountries without the deprecated countries.
+    $allCountriesOnlyAvailable = array_diff_key($allCountries, $countriesDeprecated);
+    $result = $this->checkArraysHaveSameKeys($allCountriesOnlyAvailable, $availableCountries);
```

Comment:
If we assert that the $availableCountries is equal to the array we want, that should be enough. Something like
`$this->assertEquals([
      "XK" => "XK",
      "AX" => "AX",
      "BE" => "BE",
      "TF" => "TF",
      "IT" => "IT"
    ], $availableCountries);`
This already checks that the format is correct and the deprecated country is not available

Reply:
We should also make it clear in the comments that the array needs to be keyed properly